aten/src/ATen/functorch/ADInterpreters.cpp

```diff
@@ -115,7 +116,7 @@ static void autogradBasedTransformSendToNext(
     // if (c10::show_dispatch_trace_enabled()) {
     //   std::cout << "wrap " << current_level << std::endl;
     // }
-    return makeTensorWrapper(tensor, current_level, is_immutable);
+    return makeTensorWrapper(tensor, current_level, is_immutable, life_handle);
```

Comment:
All these lower end wrappers take a const ref, for std::shared_ptr, I would recommend just passing by value and judiciously calling when passing it to the next call of makeTensorWrapper.

Reply:
```suggestion
    return makeTensorWrapper(tensor, current_level, is_immutable, std::move(life_handle));
```